### PR TITLE
use pre-commit.ci

### PR DESCRIPTION
**Pull request by PythonFZ**
Instead of using Github action runners for linting and a combination off `requirements.txt`, `pre-commit-config.yaml` and CI code we combine everything at a single place and use https://pre-commit.ci

It will also make PR to update the tools.

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -10,12 +10,9 @@ sphinx-copybutton
 IPython
 nbsphinx-link
 git+https://github.com/zincware/ZincHub.git
-isort
 coverage
 coveralls
 mdanalysis
-black
```
Comment (christophlohrmann) [inline dev-requirements.txt:17]: Ci still uses black, right?

**Review by christophlohrmann — APPROVED** [review]

**Comment by christophlohrmann** [comment]
thanks for the cleanup

**Review by SamTov — APPROVED** [review]
As long as test are passing, I am very happy with the cleanup

File: dev-requirements.txt (hunk 1 above)
Comment (PythonFZ) [reply dev-requirements.txt:17]: CI uses black, ruff, isort and everyting else that is added to the .pre-commit-config.yaml

**Comment by PythonFZ** [comment]
> As long as test are passing, I am very happy with the cleanup

I made no changes to the actual code.

**Pull request merged**